internal/client/team.go

```diff
@@ -0,0 +1,133 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+const teamBasePath = "/api/teams"
+
+type Team struct {
+	ID          int64        `json:"id"`
+	Name        string       `json:"name"`
+	Description *string      `json:"description"`
+	Members     []TeamMember `json:"members"`
+	CreatedAt   string       `json:"created_at"`
+	UpdatedAt   string       `json:"updated_at"`
+}
+
+type TeamWithoutMembers struct {
+	ID          int64   `json:"id"`
+	Name        string  `json:"name"`
+	Description *string `json:"description"`
+	CreatedAt   string  `json:"created_at"`
+	UpdatedAt   string  `json:"updated_at"`
+}
+
+type TeamMember struct {
+	UserID int64  `json:"user_id"`
+	Email  string `json:"email"`
+	Role   string `json:"role"`
+}
+
+// List of Teams
+
+type ListTeamsInput struct {
+	limit  *int
+	cursor *string
+}
+
+func (input *ListTeamsInput) SetLimit(limit int) {
+	input.limit = &limit
+}
+
+func (input *ListTeamsInput) SetCursor(cursor string) {
+	input.cursor = &cursor
+}
+
+type ListTeamsOutput struct {
+	Items      []TeamWithoutMembers `json:"items"`
+	NextCursor *string              `json:"next_cursor"`
+}
+
+const MaxListTeamsLimit = 100
+
+func (client *TroccoClient) ListTeams(input *ListTeamsInput) (*ListTeamsOutput, error) {
+	params := url.Values{}
+	if input != nil && input.limit != nil {
+		if *input.limit < 1 || *input.limit > MaxListTeamsLimit {
+			return nil, fmt.Errorf("limit must be between 1 and %d", MaxListTeamsLimit)
```

Comment:
IMO: Since the input becomes nil only when there are problems with the code. It's better to split the error handling code as follows.

```go
if input == nil {
  return nil, fmt.Errorf("input is required")
}

if *input.limit < 1 || *input.limit > MaxListTeamsLimit {
  return nil, fmt.Errorf("limit must be between 1 and %d", MaxListTeamsLimit)
}
```

Reply:
Both `limit` and `cursor` are optional, so the input can be nil without causing an error.